# light_sweep_experiments
Scripts for analyzing light sweep experiments published in Sweeney and McClean, 2023
 

Run the following scripts (in order) to generate single-cell fluorescence measurements and gene expression simulations for subsequent plotting (see plot_*.m scripts)
 
1. calc_1_run_cellfinder_in_folder_light_sweep.m
   - Loads raw images (.ND2 files) from target folders, finds/measures cells, and exports measurements (for each ND2 file) to a subfolder within the target folder

2. calc_2_collect_measurements_light_sweep.m
   - Gets gets all measurements for a given experiment (from a given target folder), labels them from a plate map (.xlsx file), and collects them into a table, which is saved a .mat file into the folder 'collected_measurements'

3. calc_3_analyze_collected_measurements_light_sweep.m
   - Loads labeled measurements from 'collected_measurements', processes data (BG quantification, irradiance correction, etc), calculates summary statistics, and exports results as large tables to .mat files for plotting

4. calc_4_run_promoter_model_LHS.m
   - Simulates expression of promoters with parameters generated by Latin hypercube sampling following the doses of Msn2 nuclear localization used in the light sweep experiments. Exports sampled parameters and simulation results to .mat files.


If starting from raw .ND2 images, start at step 1.
If starting measurements from light sweep experiments deposited to Mendeley data, start on step 2.

After running the above scripts to generate .mat files with measurements, generate plots using the following scripts:
1. plot_figures_1.m
   - Generates most figures related to light sweep experiments
2. plot_figures_2.m
   - Generates Figures S4B - S4C
3. plot_figures_3.m
   - Generates Figures 3A - 3B
4. plot_figures_4.m
   - Generates Figure S3D




